x-pack/filebeat/input/httpjson/input.go

```diff
@@ -462,6 +444,47 @@ func (in *HttpjsonInput) Wait() {
 	in.Stop()
 }
 
+func (in *HttpjsonInput) newHTTPClient(ctx context.Context) (*http.Client, error) {
+	tlsConfig, err := tlscommon.LoadTLSConfig(in.config.TLS)
+	if err != nil {
+		return nil, err
+	}
+
+	var dialer, tlsDialer transport.Dialer
+
+	dialer = transport.NetDialer(in.config.HTTPClientTimeout)
+	tlsDialer, err = transport.TLSDialer(dialer, tlsConfig, in.config.HTTPClientTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make transport client
+	client := &http.Client{
+		Transport: &http.Transport{
+			Dial:              dialer.Dial,
+			DialTLS:           tlsDialer.Dial,
+			TLSClientConfig:   tlsConfig.ToConfig(),
+			DisableKeepAlives: true,
+		},
+		Timeout: in.config.HTTPClientTimeout,
+	}
+
+	if in.config.OAuth2 == nil {
+		return client, nil
+	}
+
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, client)
+	creds := clientcredentials.Config{
+		ClientID:       in.config.OAuth2.ClientID,
+		ClientSecret:   in.config.OAuth2.ClientSecret,
+		TokenURL:       in.config.OAuth2.TokenURL,
```

Comment:
Did we want later to just add it so that tenant-id or client-id would be appended to the URL, in a case we might have a list of endpoints pre-made, that way the user would not need to know what to configure here.